src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -154,18 +161,10 @@ private void OverrideAnalysisSettings(AnalysisConfig config)
 
             if (settings.AnalyzerAssemblyPaths != null)
             {
-                AnalyzerFilePaths = settings.AnalyzerAssemblyPaths.Where(f => IsAssemblyLibraryFileName(f)).ToArray();
+                AnalyzerFilePaths = RemoveNonAnalyzerFiles(settings.AnalyzerAssemblyPaths);
             }
 
-            // Merge the additional file paths from the original list from the project with those from the
-            // config file. In case of duplicate file *names* (not full paths), use the setting from the config file.
-            var additionalFilesToRemove = GetEntriesWithMatchingFileNames(settings.AdditionalFilePaths, OriginalAdditionalFiles);
-            AdditionalFilePaths = (settings.AdditionalFilePaths ?? Enumerable.Empty<string>())
-                .Union(OriginalAdditionalFiles ?? Enumerable.Empty<string>())
-                .Except(additionalFilesToRemove)
-                .ToArray();
-
-            Log.LogMessage(MessageImportance.Low, Resources.AnalyzerSettings_RemovingDuplicateAdditionalFiles, string.Join(", ", additionalFilesToRemove) ?? "{none}");
+            AdditionalFilePaths = MergeFileLists(settings.AdditionalFilePaths, OriginalAdditionalFiles);
```

Comment:
Moved the list-merging code to a helper method, as the logic is the same for additional files and analyers.